### DISCUSSION TITLE
Download Copernicus Data Space products concurrently

### DISCUSSION THREAD

**Discussion by giorBol** [CS-SI/eodag#993] [General]
Based on the Copernicus Data Space [documentation](https://documentation.dataspace.copernicus.eu/Quotas.html#copernicus-general-users) a user is able to connect concurrently (up to 4 connections) to the provider and download the products in parallel. Is it possible to implement this while downloading from ```cop_dataspace``` provider?
As far as I know, ```download_all()```has the refresh token implemented, however it attempts to download all products **sequentially**

**Comment by sbrunato** [comment]
Hello @giorBol , there is an opened feature request for parallel downloading https://github.com/CS-SI/eodag/issues/146, but it is still not implemented
